swift-mode.el

```diff
@@ -693,6 +693,115 @@ We try to constraint those lookups by reasonable number of lines.")
                "warning: " (message) line-end))
      :modes swift-mode))
 
+;; BUILDING
+
+(defun chomp (str)
+  "Chomp leading and tailing whitespace from STR."
+  (replace-regexp-in-string (rx (or (: bos (* (any " \t\n")))
+                                    (: (* (any " \t\n")) eos)))
+                            ""
+                            str))
+
+(defun swift-mode-find-workspaces-for-directory(directory)
+  (setq command (concat "find " directory))
```

Comment:
Working on the same variables from a few function is a very bad idea. Variables supposed to hold state between function invocations should be defined in advance and their semantics should be specified. Global state is evil!